Student/SubmitAssignmentTests/SubmitAssignmentViewControllerTests.swift

```diff
@@ -0,0 +1,92 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import XCTest
+import Core
+import Social
+
+class SubmitAssignmentViewControllerTests: SubmitAssignmentTests {
+    var viewController: SubmitAssignmentViewController!
+
+    var configurationItems: [SLComposeSheetConfigurationItem]? {
+        return viewController.configurationItems() as? [SLComposeSheetConfigurationItem]
+    }
+
+    override func setUp() {
+        super.setUp()
+        LoginSession.add(.make())
+        viewController = SubmitAssignmentViewController()
+        load()
+    }
+
+    func load() {
+        XCTAssertNotNil(viewController.view)
+        env.api = URLSessionAPI()
+        env.database = database
+        env.userDefaults?.reset()
+    }
+
+    func testSelectsFirstCourseAndAssignmentByDefault() throws {
+        let presenter = try XCTUnwrap(viewController.presenter)
+        api.mock(presenter.courses, value: [.make(name: "Course 1")])
+        api.mock(GetSubmittableAssignments(courseID: "1"), value: [.make(name: "Assignment 1")])
```

Comment:
We will want to keep an eye on these to make sure this doesn't become flaky. It seemed to me that `api.mock` should have been stable but it wasn't in every case. But that could have been due to other circumstances related to specific tests.

Reply:
Thanks, Matt. I had the same concern so I ran the tests a bunch locally. I'm going to trigger CITests again to give us one more data point.